src/Validation/tests/RepoTests.cs

```diff
@@ -0,0 +1,142 @@
+            {
+                await builder.AddDefaultRepoSetupAsync();
+
+                builder.AddProject(ProjectCreator
+                    .Create()
+                    .PropertyGroup()
+                    .Property("AllowEmptySignList", "true"), "eng/Signing.props");
+
+                // Create a simple project
+                builder.AddProject(ProjectCreator
+                        .Templates
+                        .SdkCsproj(
+                            targetFramework: "net5.0",
+                            outputType: "Exe")
+                        .PropertyGroup()
+                        .Property("IsPackable", "true"),
+                    "src/FooPackage/FooPackage.csproj");
+                await builder.AddSimpleCSFile("src/FooPackage/Program.cs");
+
+                builder.Build("-configuration", "Release", "-restore", "-sign", "-ci", "-projects", "src/FooPackage/FooPackage.csproj")
+                    .Should().NotThrow();
+            }
+        }
+
+        /// <summary>
+        /// We should get an error if AllowEmptySignList is set to false, or
+        /// if it is not set at all (default behavior), and there are no items to sign.
+        /// </summary>
+        /// <param name="propertyIsSet">Is the property set or are we using the expecte ddefault?</param>
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task BuildShouldErrorIfNoItemsToSignAndNonEmptySignList(bool propertyIsSet)
+        {
+            var envVars = Environment.GetEnvironmentVariables();
+            using (var builder = new TestRepoBuilder(nameof(BuildShouldErrorIfNoItemsToSignAndNonEmptySignList), _commonRepoResourcesFixture.CommonResources))
+            {
+                await builder.AddDefaultRepoSetupAsync();
+
+                if (propertyIsSet)
+                {
+                    builder.AddProject(ProjectCreator
+                        .Create()
+                        .PropertyGroup()
+                        .Property("AllowEmptySignList", "false"), "eng/Signing.props");
+                }
+
+                // Create a simple project
+                builder.AddProject(ProjectCreator
+                        .Templates
+                        .SdkCsproj(
+                            targetFramework: "net5.0",
+                            outputType: "Exe")
+                        .PropertyGroup()
+                        .Property("IsPackable", "true"),
+                    "src/FooPackage/FooPackage.csproj");
+                await builder.AddSimpleCSFile("src/FooPackage/Program.cs");
+
+                builder.Build("-configuration", "Release", "-restore", "-sign", "-ci", "-projects", "src/FooPackage/FooPackage.csproj")
+                    .Should().Throw<Exception>().WithMessage("*error : List of files to sign is empty. Make sure that ItemsToSign is configured correctly*");
```

Comment:
Is there another way we can identify this error instead of string compare just in case the string changes in the future? (same for the other places we do this check)

Reply:
Not that I know of, but i'm open to suggestions. This is really checking for the msbuild error message. So we're checking against what the user should see.

Reply:
I suppose if this error message changes, we'll get an error on this test and can address it then. Was hoping that maybe we could look for a specific exception type, but not sure if that's something we can do with MSBuild stuff. No big deal either way.